vaadin-list-mixin.html

```diff
@@ -208,18 +210,27 @@
       return -1;
     }
 
+    _setHidden(item) {
+      if (getComputedStyle(item).getPropertyValue('display') === 'none') {
+        item.hidden = true;
+      }
+    }
+
     _setFocusable(idx) {
-      idx = this._getAvailableIndex(idx, 1, item => !item.disabled);
+      idx = this._getAvailableIndex(idx, 1, item => !(item.disabled || getComputedStyle(item).getPropertyValue('display') === 'none'));
       const item = this.items[idx] || this.items[0];
       this.items.forEach(e => e.tabIndex = e === item ? 0 : -1);
     }
 
     _focus(idx) {
       const item = this.items[idx];
-      this.items.forEach(e => e.focused = e === item);
+      this._setHidden(item);
+      this.items.forEach(e => e.focused = (e === item && !(e.hidden || e.disabled)));
       this._setFocusable(idx);
-      this._scrollToItem(idx);
-      item.focus();
+      if (!(item.hidden || item.disabled)) {
```

Comment:
Rather than fixing the logic in `_focus` it might be better to do it where `_focus` gets invoked (in `_onKeydown`, like the previous fix)

Reply:
done